nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -437,20 +454,17 @@ bool DockingServer::approachDock(Dock * dock, geometry_msgs::msg::PoseStamped &
         tf2::getYaw(target_pose.pose.orientation) + M_PI);
     }
 
-    // The control law can get jittery when close to the end when atan2's can explode.
-    // Thus, we backward project the controller's target pose a little bit after the
-    // dock so that the robot never gets to the end of the spiral before its in contact
-    // with the dock to stop the docking procedure.
-    const double backward_projection = 0.25;
-    const double yaw = tf2::getYaw(target_pose.pose.orientation);
-    target_pose.pose.position.x += cos(yaw) * backward_projection;
-    target_pose.pose.position.y += sin(yaw) * backward_projection;
-    tf2_buffer_->transform(target_pose, target_pose, base_frame_);
-
     // Compute and publish controls
     auto command = std::make_unique<geometry_msgs::msg::TwistStamped>();
-    command->header.stamp = now();
-    if (!controller_->computeVelocityCommand(target_pose.pose, command->twist, dock_backwards_)) {
+
+    geometry_msgs::msg::PoseStamped robot_pose = getRobotPoseInFrame(target_pose.header.frame_id);
+    const double angle_to_goal = angles::shortest_angular_distance(
+    tf2::getYaw(robot_pose.pose.orientation), atan2(robot_pose.pose.position.y - target_pose.pose.position.y, robot_pose.pose.position.x - target_pose.pose.position.x));
+
+    if(initial_rotation_ && fabs(angle_to_goal) > initial_rotation_min_angle_){
+      command->twist = controller_->rotateToTarget(angle_to_goal);
```

Comment:
It seems more natural to have the initial rotation bit performed here: https://github.com/ros-navigation/navigation2/blob/14faf822a6595293cb045b064b6cde867d0731d5/nav2_docking/opennav_docking/src/docking_server.cpp#L288

For example:

```
    while (rclcpp::ok()) {
      try {
        // Invert robot orientation if necessary
        if (dock_backward_without_sensor_) {
            rotate180();
        }
        // Approach the dock using control law
        if (approachDock(dock, dock_pose)) {
          // We are docked, wait for charging to begin
          RCLCPP_INFO(
            get_logger(), "Made contact with dock, waiting for charge to start (if applicable).");

```

Then the rest stays the same